apps/pie-storybook/stories/testing/pie-link.test.stories.ts

```diff
@@ -0,0 +1,250 @@
+            control: 'text',
+        },
+    },
+    args: defaultArgs,
+};
+
+export default linkStoryMeta;
+
+const Template : TemplateFunction<LinkProps> = ({
+    aria,
+    tag,
+    href,
+    target,
+    rel,
+    size,
+    variant,
+    underline,
+    type,
+    isBold,
+    isStandalone,
+    hasVisited,
+    slot,
+    iconPlacement,
+}) => html`
+        <pie-link
+            .aria="${aria}"
+            tag="${ifDefined(tag)}"
+            variant="${ifDefined(variant)}"
+            size="${ifDefined(size)}"
+            underline="${ifDefined(underline)}"
+            iconPlacement="${ifDefined(iconPlacement)}"
+            href="${ifDefined(href)}"
+            target="${ifDefined(target)}"
+            rel="${ifDefined(rel)}"
+            type="${ifDefined(type)}"
+            ?isBold="${isBold}"
+            ?isStandalone="${isStandalone}"
+            ?hasVisited="${hasVisited}">
+            ${iconPlacement ? html`<icon-plus-circle slot="icon"></icon-plus-circle>` : nothing}
+            ${sanitizeAndRenderHTML(slot)}
+        </pie-link>`;
+
+const createLinkStory = createStory<LinkProps>(Template, defaultArgs);
+
+export const Default = createLinkStory();
+export const HighVisibility = createLinkStory({ variant:  'high-visibility' });
+export const Inverse = createLinkStory({ variant: 'inverse' }, { bgColor: 'dark (container-dark)' });
+
+const VariantsTemplate: TemplateFunction<LinkProps> = ({
+    variant,
+    size,
+    isBold,
+    iconPlacement,
+    isStandalone,
+    tag,
+}) => html`
+    <div style="background-color: ${variant === 'inverse' ? 'var(--dt-color-container-dark)' : 'transparent'}">
+        <pie-link
+            variant="${ifDefined(variant)}"
+            size="${ifDefined(size)}"
+            iconPlacement="${ifDefined(iconPlacement)}"
+            tag="${ifDefined(tag)}"
+            ?isBold="${isBold}"
+            ?isStandalone="${isStandalone}">
+            ${iconPlacement ? html`<icon-plus-circle slot="icon"></icon-plus-circle>` : nothing}
+            Link Text
+        </pie-link>
+    </div>
+`;
+
+// Base shared props without variant or size
+const baseSharedPropsMatrix = {
+    tag: [...tags],
+    isBold: [true, false],
+    iconPlacement: [undefined, ...iconPlacements],
+    isStandalone: [true, false],
+};
+
+// Default variant stories
+const defaultSmallPropsMatrix = {
+    ...baseSharedPropsMatrix,
+    variant: ['default'],
+    size: ['small'],
+};
+
+const defaultMediumPropsMatrix = {
+    ...baseSharedPropsMatrix,
+    variant: ['default'],
+    size: ['medium'],
+};
+
+export const DefaultSmallVariations = createVariantStory(VariantsTemplate, defaultSmallPropsMatrix);
+export const DefaultMediumVariations = createVariantStory(VariantsTemplate, defaultMediumPropsMatrix);
+
+// High-visibility variant stories
+const highVisibilitySmallPropsMatrix = {
+    ...baseSharedPropsMatrix,
+    variant: ['high-visibility'],
+    size: ['small'],
+};
```

Comment:
Can we leverage TS Types for the matrix objects?

You can see an example in pie-card - https://github.com/justeattakeaway/pie/blob/main/apps/pie-storybook/stories/testing/pie-card.test.stories.ts#L168-L199